src/x509.c

```diff
@@ -3078,19 +3087,25 @@ char* wolfSSL_X509_get_name_oneline(WOLFSSL_X509_NAME* name, char* in, int sz)
             * Add extra strSz for '=' and '\0' characters in XSNPRINTF.
             */
             strSz = (int)XSTRLEN(sn) + nameSz + 2;
-            totalSz+= strSz;
             str = (char*)XMALLOC(strSz, NULL, DYNAMIC_TYPE_TMP_BUFFER);
             if (str == NULL) {
                 WOLFSSL_MSG("Memory error");
                 return NULL;
             }
-            if (XSNPRINTF(str, strSz, "%s=%s", sn, buf) >= strSz) {
+            if ((str_len = XSNPRINTF(str, strSz, "%s=%s", sn, buf)) >= strSz) {
                 WOLFSSL_MSG("buffer overrun");
+                XFREE(str, NULL, DYNAMIC_TYPE_TMP_BUFFER);
                 return NULL;
             }
         }
         /* Copy string to tmpBuf */
-        XSTRNCAT(tmpBuf, str, strSz);
+        if (totalSz + str_len > (int)sizeof(tmpBuf)) {
```

Comment:
will totalSz always be 0 here? the github gui is showing the two spots where totalSz is incremented as being removed

Reply:
it's now incremented in a new spot:
```
        XMEMCPY(tmpBuf + totalLen, str, strLen);
        totalLen += strLen;
 ```
(note I'm also renaming `totalSz` as `totalLen` for clarity -- in this file, `Sz` iis used to mean the allocation size, and `Len` is the used length within that allocation.)